pkg/controlplane/apiserver/config.go

```diff
@@ -311,10 +316,17 @@ func CreateConfig(
 		if err != nil {
 			return nil, nil, err
 		}
+
+		discoveryScheme := runtime.NewScheme()
```

Comment:
Seems a bit weird we're constructing this here. Can we create it deeper in the peerproxy initialization? That way we don't need an extra parameter in the BuildPeerProxy initialization.